src/main/java/aquality/appium/mobile/screens/Screen.java

```diff
@@ -24,47 +22,27 @@
     protected Screen(By locator, String name) {
         this.locator = locator;
         this.name = name;
-        screenLabel = getElementFactory().getLabel(locator, name);
+        this.screenLabel = getElementFactory().getLabel(locator, name);
     }
 
-    protected abstract AppiumDriver getDriver();
-
-    /**
-     * Return screen state for screen locator
-     *
-     * @return True - screen is opened,
-     * False - screen is not opened
-     */
-    public boolean isDisplayed() {
-        return isDisplayed(null);
+    protected IElementFactory getElementFactory(){
+        return AqualityServices.getElementFactory();
     }
 
-    /**
-     * Return screen state for screen locator
-     *
-     * @param timeout timeout for action
-     * @return True - screen is opened,
-     * False - screen is not opened
-     */
-    public boolean isDisplayed(Duration timeout) {
-        return screenLabel.state().waitForDisplayed(timeout);
+    @SuppressWarnings("unchecked")
+    protected T getDriver() {
```

Comment:
do we really need to return driver directly from the Screen, not from the AqualityServices?

Reply:
Yes, because this is driver casted to it's actual type (IOSDriver or AndroidDriver), and they have different interfaces inside. So if we want to perform smth custom with the driver (like start an activity for android or send a command to simulator on iOS), we should be doing it at screenObjects level. The getter to casted driver is needed here for this purpose

Reply:
Not sure. @mialeska what do you think? Btw, it is protected, not public.